var/spack/repos/builtin/packages/steps/package.py

```diff
@@ -0,0 +1,108 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os.path as osp
+
+from spack import *
+
+
+class Steps(CMakePackage):
+    """STochastic Engine for Pathway Simulation"""
+
+    homepage = "http://www.example.com"
+    git      = "https://github.com/CNS-OIST/STEPS.git"
+
+    version("3.3.0", submodules=True)
+    version("3.2.0", submodules=True)
+    version("develop", branch="master", submodules=True)
+
+    variant("native", default=True, description="Generate non-portable arch-specific code")
+    variant("bindings", default="cython",
+            values=("cython", "swig"),
+            description="Generate bindings using Cython, instead of Swig")
+    variant("lapack", default=False, description="Use new BDSystem/Lapack code for E-Field solver")
+    variant("petsc", default=False, description="Use PETSc library for parallel E-Field solver")
+    variant("mpi", default=True, description="Use MPI for parallel solvers")
+
+    depends_on("py-cython", when="bindings=cython")
+    depends_on("swig", when="bindings=swig")
+    depends_on("lapack", when="+lapack")
+    
+    depends_on("petsc^debug+int64", when="+petsc")
+    # @pramodk: I have spec conflict when I use python 3: spack spec steps%gcc+petsc^python@3
+    # because petsc still uses Python 2
+    # I tried to replace line above by:
+    #
+    # depends_on("petsc^debug+int64^python@3", when="+petsc^python3")
+    # depends_on("petsc^debug+int64^python@2", when="+petsc^python2")
+    #
+    # but `when` directive is always evaluate to False, whether I put python@2 or python@3
+    # please help!
+
+    depends_on("mpi", when="+mpi")
+    depends_on("blas")
+
+    def cmake_args(self):
+        args = []
+        spec = self.spec
+        bindings =  spec.variants["bindings"].value
+
+        if "+native" in spec:
+            args.append("-DTARGET_NATIVE_ARCH:BOOL=True")
+        else:
+            args.append("-DTARGET_NATIVE_ARCH:BOOL=False")
+
+        if bindings == "cython":
+            args.append("-DPYTHON_BINDINGS_CYTHON:BOOL=True")
+        elif bindings == "swig":
+            args.append("-DPYTHON_BINDINGS_CYTHON:BOOL=False")
+        else:
+            raise Exception("Unexpected bindings type: " + bindings)
```

Comment:
raising and exception is not required, already taken care by spack

Reply:
This exception is more for maintainers, in case the variant is updated but not this hunk.

Reply:
Ok